WalletWasabi.Gui/Controls/WalletExplorer/WalletViewModelBase.cs

```diff
@@ -0,0 +1,88 @@
+using ReactiveUI;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using WalletWasabi.Gui.ViewModels;
+using WalletWasabi.Helpers;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Gui.Controls.WalletExplorer
+{
+	public class WalletViewModelBase : ViewModelBase, IComparable<WalletViewModelBase>
+	{
+		private bool _isExpanded;
+		private bool _isBusy;
+		private string _title;
+		private WalletState _walletState;
+
+		public WalletViewModelBase(Wallet wallet)
+		{
+			Wallet = Guard.NotNull(nameof(wallet), wallet);
+
+			Wallet = wallet;
+			Title = WalletName;
+
+			WalletState = wallet.State;
+
+			this.WhenAnyValue(x => x.WalletState)
+				.Subscribe(x =>
+				{
+					switch (x)
+					{
+						case WalletState.Initialized:
+						case WalletState.Started:
+						case WalletState.Stopped:
+						case WalletState.Uninitialized:
+							IsBusy = false;
+							break;
+
+						default:
+							IsBusy = true;
+							break;
+					}
+				});
+		}
+
+		public WalletState WalletState
```

Comment:
How about forwarding this from the Model with ToProperty?

Reply:
Probably ToProperty is not good because we do not use INotifyProperty. How is WalletState updated here?

Reply:
Its updated here:
https://github.com/zkSNACKs/WalletWasabi/blob/multiwallet-wallet-explorer/WalletWasabi.Gui/Controls/WalletExplorer/WalletExplorerViewModel.cs#L47

Reply:
viewmodels follow the design of the models, 1 to 1 relationship, so the event subscription moved from outside WalletViewModel to WalletExplorerViewModel (which is the equivalent of WalletManager class)

Reply:
Subscribes internally... much better!